packages/mail/NewsletterSubscription.js

```diff
@@ -0,0 +1,46 @@
+const intersect = (arr1, arr2) => [...new Set(arr1)].filter(num => new Set(arr2).has(num))
+
+class NewsletterSubscription {
+  constructor (userId, interestId, subscribed, roles) {
+    const interstConfiguration = this.constructor.interestConfiguration(interestId)
+    this.name = interstConfiguration.name
+    this.id = Buffer.from(userId + this.name).toString('base64')
+    this.subscribed = subscribed
+    this.isEligible = this.constructor.isEligibleForInterestId(interestId, roles)
+  }
+
+  static configure (config) {
+    this.interestConfigurationMap = config
+  }
```

Comment:
globals need to be clearly visible, this is not the case here. For a developer who needs to change something about the subscriptions, who doesn't have the context of this PR, would have a hard time figure out that this is a class global that needs to be set first.

Reply:
the more I think about it, a class is the wrong approach here. You would need to check in all methods if configure was called before (precondition-check), I would instead turn all class functions, into normal functions that take a config and are stateless.